src/contexts/AuthUserContext/index.tsx

```diff
@@ -0,0 +1,40 @@
+import { User as AuthUser } from '@firebase/auth-types';
+import React, { createContext, ReactElement, useContext, useEffect, useState } from 'react';
+
+import { useFirebase } from '../../contexts/FirebaseContext';
+
+/**
+ * Holds the most up-to-date context object.
+ */
+const AuthUserContext = createContext<AuthUser | null>(null);
+
+/**
+ * @component AuthUserProvider
+ * The context provider component for an auth user.
+ *
+ * @param children - The child components or elements
+ */
+export function AuthUserProvider({
+  children,
+}: {
+  children: ReactElement | ReactElement[];
+}): ReactElement {
+  const [authUser, setAuthUser] = useState<AuthUser | null>(null);
+  const firebase = useFirebase();
+
+  useEffect(() => {
```

Comment:
I just want to ask you about `onAuthStateChanged`. Is this called every time the user `signIn` or `signUp` ?

Reply:
onAuthStateChanged is called every time the user signs in, but the user will automatically be signed in after signing up for an account too

Reply:
Sounds good!